wunderio/drupal-project
# #357: Default configuration for Varnish Purge

This delivers https://github.com/wunderio/drupal-project/issues/357.

### Contains
- Added Purge and Varnish Purge modules to `composer.json` 
- Updated the very old Basic installation profile to support D10
  - Export bunch of new and updated configurations from a fresh install
  - Added Purge and Varnish purge module configurations to the profile
- Added the `VARNISH_ALLOW_UNRESTRICTED_PURGE` to Lando Varnish configuration for allowing purges
- Add Varnish Purger configuration to settings.php. These configuration overrides apply only for the Varnish Purger instance that's configured in the 'Basic' installation profile as they have unique configuration names (`varnish_purger.settings.f94540554c`)
- Updated README with how to set up Varnish + Purge for new and existing sites.

### Testing
- Uncomment Lando Varnish configuration (`services` → `varnish` and `proxy` → `varnish`)
- Rebuild Lando: `lando rebuild -y`
- Install site with the Basic profile: `lando drush si basic -y`
- Create some basic page contents (https://drupal-project.lndo.site/node/add/page)
- Visit the created pages in the Varnish host (https://varnish.drupal-project.lndo.site/node/1)
- Confirm that the `X-W-Cache: HIT` and `X-W-Cache-Hits: N` headers are present and working
- Edit the created pages and save them
- See that Varnish has purged that page and that the changes are visible
